### ISSUE TITLE
What is the result for 5epochs.

### ISSUE THREAD

**Issue by mx8435** [llamazing/numnet_plus#3]
Hi, what is the final `f1` and `em` for `sh train.sh 345 5e-4 1e-4 5e-4 0.01` ?

**Comment by llamazing** [comment]
Sorry, we have not tested the group of hyperparameters. 
For `sh train.sh 345 5e-4 1.5e-5 5e-5 0.01` we get `f1 82.19` and `em 78.57`.
To facilitate the reproduction of the results, we have changed the readme file for the hyperparameters.

**Comment by mx8435** [comment]
> Sorry, we have not tested the group of hyperparameters.
> For `sh train.sh 345 5e-4 1.5e-5 5e-5 0.01` we get `f1 82.19` and `em 78.57`.
> To facilitate the reproduction of the results, we have changed the readme file for the hyperparameters.

Are `f1 82.19` and `em 78.57` the best dev metrics? And use 5 epochs? 
Besides, would you use multi-gpus for training? I found it's time-consuming.

**Comment by llamazing** [comment]
> > Sorry, we have not tested the group of hyperparameters.
> > For `sh train.sh 345 5e-4 1.5e-5 5e-5 0.01` we get `f1 82.19` and `em 78.57`.
> > To facilitate the reproduction of the results, we have changed the readme file for the hyperparameters.
> 
> @llamazing Thanks for your reply. :)
> Are `f1 82.19` and `em 78.57` the best dev metrics? And use 5 epochs?
> Besides, are you use multi-gpus when training? I found it's time-consuming when using single gpu.

It may not be the best dev metric. We did not spend too much time on hyperparameter selection.
The code does not support multi-GPUs training now.

**Comment by mx8435** [comment]
> > > Sorry, we have not tested the group of hyperparameters.
> > > For `sh train.sh 345 5e-4 1.5e-5 5e-5 0.01` we get `f1 82.19` and `em 78.57`.
> > > To facilitate the reproduction of the results, we have changed the readme file for the hyperparameters.
> > 
> > 
> > @llamazing Thanks for your reply. :)
> > Are `f1 82.19` and `em 78.57` the best dev metrics? And use 5 epochs?
> > Besides, are you use multi-gpus when training? I found it's time-consuming when using single gpu.
> 
> All our experiments use 5 epochs and we did not spend too much time on hyperparameter selection. So it may not be the best dev metric.
> The code does not support multi-GPUs training now.

I've ran `sh train.sh 345 5e-4 1.5e-5 5e-5 0.01` but only get:

Eval 9536 examples, result in epoch 5, eval loss 377360.05822300387, eval em 0.747378355704698 eval f1 0.7834899328859062.
model saved to ./numnet_plus_345_LR_5e-4_BLR_1.5e-5_WD_5e-5_BWD_0.01/checkpoint_best

I never change any parameter or file. Can you help me to reproduce your result.

**Comment by llamazing** [comment]
Is your `pytorch_transformer` version is `1.1.0`?
Below is our training logs with `345 5e-4 1.5e-5 5e-5 0.01`.
>> 09/23/2019 08:29:40  Eval 9536 examples, result in epoch 1, eval loss 215934.20358306728, eval em 0.7159186241610739 eval f1 0.7502957214765109.
09/24/2019 05:25:21 Eval 9536 examples, result in epoch 2, eval loss 215934.00826157143, eval em 0.7624790268456376 eval f1 0.7970060822147659.
09/24/2019 02:21:21 Eval 9536 examples, result in epoch 3, eval loss 215934.07345209384, eval em 0.7703439597315436 eval f1 0.804992659395974.
09/24/2019 11:17:03 Eval 9536 examples, result in epoch 4, eval loss 215934.30083180242, eval em 0.7857592281879194 eval f1 0.8197472734899331.
09/25/2019 08:12:32 Eval 9536 examples, result in epoch 5, eval loss 215934.29386250713, eval em 0.7856543624161074 eval f1 0.8198468959731549.

We have done another experiment to check this. Please wait for some time.

**Comment by mx8435** [comment]
> pytorch_transformer

@llamazing
Yes, I'm using `1.1.0`
#pip list|grep transformer
pytorch-transformers          1.1.0

**Comment by llamazing** [comment]
> > pytorch_transformer
> 
> @llamazing
> Yes, I'm using `1.1.0`
> #pip list|grep transformer
> pytorch-transformers 1.1.0
> 
> Besides, I found that the config `--gradient_accumulation_steps 4` in `TRAIN_CONFIG` has been changed to 8, since the GPU I used is 16G. But I think it will not affect the training result?
> No other parameter is changed.

We run our experiments again and it seems correct in 3 epoch.
Below is our training log.
> 10/26/2019 10:24:32 Eval 9536 examples, result in epoch 1, eval loss 215934.2553895517, eval em 0.7307046979865772 eval f1 0.7657005033557055.
10/27/2019 05:31:11 Eval 9536 examples, result in epoch 2, eval loss 215934.08169940358, eval em 0.7555578859060402 eval f1 0.792539848993289.
10/27/2019 12:34:20 Eval 9536 examples, result in epoch 3, eval loss 215934.0072322501, eval em 0.7729656040268457 eval f1 0.8091086409395983.

I'm not sure whether `gradient_accumulation_steps` will affect the result.
